packages/extension-polkagate/src/hooks/useValidatorApy.ts

```
@@ -0,0 +1,128 @@
+    const actualDepth = validatorEraInfo.length;
+
+    totalStaked = totalStaked.div(decimal).divn(actualDepth);
+
+    const dailyReward = totalRewards.div(decimal).divn(actualDepth);
+
+    // Calculate daily return as a fraction of the staked amount
+    const dailyReturn = dailyReward.toNumber() / totalStaked.toNumber();
+
+    const APY = (dailyReturn * 365 * 100).toFixed(2);
+
+    if (!isFinite(+APY) || isNaN(+APY)) {
```

Biome

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)